src/rx5808-pro-diversity/state.cpp

```diff
@@ -0,0 +1,112 @@
+#include <stddef.h>
+
+#include "state.h"
+
+#include "state_screensaver.h"
+#include "state_search.h"
+#include "state_bandscan.h"
+#include "state_menu.h"
+#include "state_settings.h"
+#include "state_settings_rssi.h"
+
+#include "ui.h"
+#include "buttons.h"
+
+#include "timer.h"
+
+
+void *operator new(size_t size, void *ptr){
+  return ptr;
+}
+
+#define MAX(a, b) (a > b ? a : b)
+#define STATE_BUFFER_SIZE \
+    MAX(sizeof(ScreensaverStateHandler), \
+    MAX(sizeof(SearchStateHandler), \
+    MAX(sizeof(BandScanStateHandler), \
+    MAX(sizeof(MenuStateHandler), \
+    MAX(sizeof(SettingsStateHandler), \
+        sizeof(SettingsRssiStateHandler) \
+    )))))
+;
+
+namespace StateMachine {
+    static void onButtonChange(Button button, Buttons::PressType pressType);
+    static StateHandler *getStateHandler(State stateType);
+
+
+    static uint8_t stateBuffer[STATE_BUFFER_SIZE];
```

Comment:
The search state must be the biggest one at the moment (I think). From a quick look around it looks like there's no easy way to spit out the sizes but I should be able to do it manually.

It's calculated based on the size of the largest state so it should be exactly as large as it needs to be. States get swapped in and out of this buffer as they need it.

Since I implemented this I've not ran into any huge memory issues but I'm sure they could still be some improvement. The logged RSSI values for the graph for example take up a ton of space.